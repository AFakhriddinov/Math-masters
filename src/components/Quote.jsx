import React, { useState, useEffect } from 'react';

export default function Quote() {
  const [quoteData, setQuoteData] = useState([]);
  const [isLoading, setIsLoading] = useState(false);
  const [hasError, setHasError] = useState(false);
  const style = {
    textAlign: 'center',
  };
  useEffect(() => {
    const fetchQuote = async () => {
      setIsLoading(true);
      try {
        const response = await fetch(
          'https://api.api-ninjas.com/v1/quotes?category=education',
          {
            method: 'GET',
            withCredentials: true,
            headers: {
              'X-Api-Key': 'fi+gQd6ALdiLpzmjNljl2w==PAE0GpskRB6t84mB',
              'Content-Type': 'application/json',
            },
          },
        );
        const json = await response.json();
        setQuoteData(json);
      } catch (error) {
        setHasError(true);
      }
      setIsLoading(false);
    };
    fetchQuote();
  }, [setQuoteData, setIsLoading]);
  if (hasError) return <div style={style}>Quote could not be found</div>;
  if (isLoading) return <div style={style}>Loading...</div>;
  return (
    <div style={style}>
      {quoteData.map((quote) => (
        <p key={quote.quote}>
          {quote.quote}
          <br />
          <b>{quote.author}</b>
        </p>
      ))}
    </div>
  );
}
