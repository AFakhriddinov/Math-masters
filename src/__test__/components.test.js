import React from "react";
import { render } from "@testing-library/react";
import Home from "../components/Home";
import Quote from "../components/Quote";

describe("Testing Home, Quote, and Navigation components render correctly", () => {
  it("Rendering Home component", () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });

  
});