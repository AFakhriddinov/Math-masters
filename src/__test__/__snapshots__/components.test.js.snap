// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Home, Quote, and Navigation components render correctly Rendering Home component 1`] = `
<DocumentFragment>
  <div
    class="home"
  >
    <h2>
      Welcome to our page
    </h2>
    <p>
      This technology allows students solve complicated problems quickly and in an efficient manner. Additionally, it can reduce the problem to simpler tasks and allows the student to devote more time in understanding the problem. Secondly, they are saved from monotonous calculations and the same boring mundane procedure.
    </p>
    <p>
      The above advantages help students to avoid boredom, and it does not demoralize their mathematical understanding. If mathematics is not made entertaining, pupils can feel bored, and it can demoralize their mathematical understanding. The importance of this helpful tool was known in the early times. At that time, early men use bones, pebbles and other objects to tally the calculations. But, it was very limited in nature. So, when the first mainframe computers came into existence. It was using vacuum tubes and transistors. That leads to stepping stones to develop an electronic calculator.
    </p>
  </div>
</DocumentFragment>
`;
